lib/services/build.ts

```diff
@@ -317,8 +318,11 @@ export class BuildService implements Project.IBuildService {
 			this.$jsonSchemaValidator.validate(this.$project.projectData);
 			this.$jsonSchemaValidator.validateWithBuildSchema(this.$project.projectData, settings.platform);
 
-			settings.configuration = settings.configuration || (this.$options.release ? "Release" : "Debug");
-			this.$logger.info("Building project for platform '%s', configuration '%s'", settings.platform, settings.configuration);
+			settings.projectConfiguration = settings.projectConfiguration || (this.$options.config && this.$options.config[0]) || constants.Configurations.Debug;
```

Comment:
If `this.$options.config[0]` is invalid project configuration the build will fail with `Cannot read property 'AppIdentifier' of undefined`. Check if the configuration is valid and if it is not valid - throw an error with suitable message.